### PR TITLE
ENT-4071: Updated tasks to support Braze

### PR DESCRIPTION
**Pull request by irfanuddinahmad**
**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production
    - `test.in` for test requirements
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] [Version](https://github.com/edx/ecommerce-worker/blob/master/setup.py) bumped

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/ecommerce-worker/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [Travis](https://travis-ci.org/github/edx/ecommerce-worker), verify version has been pushed to [PyPI](https://pypi.org/project/edx-ecommerce-worker/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [ecommerce](https://github.com/edx/ecommerce) to upgrade dependencies (including ecommerce-worker)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in ecommerce will look for the latest version in PyPi.
    - Note: the ecommerce-worker constraint in ecommerce **must** also be bumped to the latest version in PyPi.
- [ ] Deploy `ecommerce`
- [ ] Deploy `ecomworker` on GoCD.
    - While some functions in ecommerce-worker are run via ecommerce, others are handled by a standalone AMI and must be
      released via GoCD.

### CODE REVIEW

**Review by moconnell1453 — APPROVED** [review]

File: ecommerce_worker/configuration/base.py [hunk 1]
```diff
@@ -78,7 +78,10 @@
     'BRAZE_WEBAPP_API_KEY': None,
     'REST_API_URL': 'https://rest.iad-06.braze.com',
     'MESSAGES_SEND_ENDPOINT': '/messages/send',
-    'FROM_EMAIL': '<edx-for-business-no-reply@info.edx.org>'
+    'FROM_EMAIL': '<edx-for-business-no-reply@info.edx.org>',
+    # Retry settings for Sailthru celery tasks
```
Comment (muhammad-ammar) [inline ecommerce_worker/configuration/base.py]: I think comment needs to be updated?

File: ecommerce_worker/sailthru/v1/tasks.py [hunk 2]
```diff
@@ -475,7 +580,57 @@ def send_offer_update_email(self, user_email, subject, email_body, sender_alias,
...
+        site_code (str): Identifier of the site sending the email.
+    """
+    config = get_braze_configuration(site_code)
+    try:
+        braze_client = get_braze_client(site_code)
+        braze_client.send_message(
+            list(emails),
+            subject,
+            email_body,
+            ''
```
Comment (sameenfatima78) [inline ecommerce_worker/sailthru/v1/tasks.py]: Would be nice to use keyword arguments here and everywhere else in the code

**Review by sameenfatima78 — APPROVED** [review]
Just one suggestion

**Review by muhammad-ammar — APPROVED** [review]
LGTM

**Review by saleem-latif — APPROVED** [review]

File: ecommerce_worker/sailthru/v1/tasks.py (hunk 2 above)
Comment (irfanuddinahmad) [reply ecommerce_worker/sailthru/v1/tasks.py]: Done

**Pull request merged**